### PR TITLE
Fix for /DIESEL ENGINE search triggering /DIE command and related 

### PR DESCRIPTION
**Pull request by Riik**
Fix for https://github.com/twohoursonelife/OneLife/issues/154
**TL;DR** Make commands only trigger on exact match, instead of when command occurs at beginning of typedText 

**What's causing this**
When a command is entered, the is a line to check if it matches the `/DIE` command: 
```if( strstr( typedText, translate( "dieCommand" ) ) == typedText )```
`strstr` check that the command occurs in typedText, and returns the beginning of the match. Comparing the return value with *typedText ensures the match is at the start of typedText. Any texts that comes after this is ignored 

**Jason's fix**
Jason fixed this in this commit: https://github.com/jasonrohrer/OneLife/commit/de536031c4cb979e28d523a1817d444869dc12ef
The commit introduces a new function `commandTyped` that does the following: 
  - Executes the same `strstr` check
  - Trims whitespace from typedText
  - Check that trimmed typedText has the same length as the command text 

**Implementation in this PR**
I replaced the `strstr` substring search with `strcmp` instead of adding the extra `commandTyped` function. This makes sure the command only gets triggered when there is an exact match between the command and the typed text. I added a similar fix to `getEmotionIndex` for `/SADDLE` clashing with `/SAD`, which makes emotions also trigger only on exact matches. If I missed something I'd also be happy to implement the `commandTyped` function instead, but this seemed more straightforward to me. 

**Testing**
I've compiled and tested locally, and checked that `/DIESEL ENGINE` and `/SADDLE` now return the correct search results, and that `/SAD` and `/DIE` also still work. I've tested some other random emotions and verified `/DISCONNECT`, `/PING`, `/FPS` and `/NETWORK` also still work. 

**Remarks**
  - Since we're testing for an exact match, a command with a whitespace at the end will also not match. If we want a bit more flexibility we could call `trimWhitespace` before the `strcmp`.
  - `getEmotionIndex` also changed from match at the beginning to exact string match. It's also used in `EditorScenePage.cpp:644`. I'm not sure if that might cause trouble.

### CODE REVIEW

**Review by risvh — APPROVED** [review]

**Comment by connorhsm** [comment]
@Riik Thank you for this contribution! I greatly appreciate your well explained notes. Hope to see you around more!

**Comment by Riik** [comment]
Thanks Connor! Was fun making this work and happy that it's merged! Few busy weeks ahead but will keep an eye open for the issues page, interested to pick up more tasks in the future.

**Comment by jxng1** [comment]
Hey, new to the codebase and mainly just poking my head into things.

**Looks good!**

Yes, with your point on non-matches, it'd be good to always trim leading/trailing whitespaces in command calls; although trimWhitespace() would not work as it'd trim all whitespaces - including between multi-word commands, much like **/DIESEL ENGINE**.

There are multiple ways to implement that; you've got this 🦾 

Do let me know if my comment process are wrong/if there is a guideline for review comments(I had a dig and didn't find anything concrete). This is the first time in a while 😓

**Comment by Riik** [comment]
Hey @jxng1, thanks for looking at the merge request, and thanks for your feedback. 
The merge request has already been merged in the mean time, so I think it's fine to leave it as-is for now. With the current implementation it should match only `/DIE` exactly, and not `/DIE ` with an additional whitespace at the end. 

Regarding your command about in-between whitespace, if we were to add the trimWhiteSpace I don't think that would be an issue - since the conversion only would happen when checking for a command. Afaik all the commands (/DIE, /DISCONNECT, /PING, /FPS and /NETWORK) and emotions don't have spaces in them. But again, I think it's probably fine to only have exact matches, and have it not match if there is a trailing whitespace.
